File: src/main/java/org/redrune/net/packet/context/impl/ItemOnItemPacketContext.java
package org.redrune.net.packet.context.impl;

import org.redrune.game.content.entity.actor.player.event.item.ItemInterfaceInteractionEvent;
import org.redrune.game.entity.actor.player.Player;
import org.redrune.net.packet.context.PacketContext;

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 2019-02-04
 */
public class ItemOnItemPacketContext extends PacketContext {
	
	private final int interfaceId;
	
	private final int itemUsedId;
	
	private final int fromSlot;
	
	private final int interfaceId2;
	
	private final int itemUsedWithId;
	
	private final int toSlot;
	
	public ItemOnItemPacketContext(int interfaceId, int itemUsedId, int fromSlot, int interfaceId2, int itemUsedWithId, int toSlot) {
		this.interfaceId = interfaceId;
		this.itemUsedId = itemUsedId;
		this.fromSlot = fromSlot;
		this.interfaceId2 = interfaceId2;
		this.itemUsedWithId = itemUsedWithId;
		this.toSlot = toSlot;
	}
	
	@Override
	public void handle(Player player) {
		player.getEventManager().start(new ItemInterfaceInteractionEvent(interfaceId, itemUsedId, fromSlot, interfaceId2, itemUsedWithId, toSlot));
	}
}

File: src/main/java/org/redrune/networking/packet/context/impl/ItemOnItemPacketContext.kt
package org.redrune.networking.packet.context.impl

import org.redrune.game.content.entity.actor.player.event.item.ItemInterfaceInteractionEvent
import org.redrune.game.entity.actor.player.Player
import org.redrune.networking.packet.context.PacketContext

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 2019-02-04
 */
class ItemOnItemPacketContext(
    private val interfaceId: Int,
    private val itemUsedId: Int,
    private val fromSlot: Int,
    private val interfaceId2: Int,
    private val itemUsedWithId: Int,
    private val toSlot: Int
) : PacketContext() {
    override fun handle(player: Player) {
        player.eventManager.start(
            ItemInterfaceInteractionEvent(
                interfaceId,
                itemUsedId,
                fromSlot,
                interfaceId2,
                itemUsedWithId,
                toSlot
            )
        )
    }
}